src/snowflake/snowpark/session.py

```diff
@@ -1856,11 +1858,23 @@ def sproc(self) -> StoredProcedureRegistration:
         """
         return self._sp_registration
 
+    def _infer_is_return_table(self, sproc_name: str, *args: Any) -> bool:
+        try:
+            arg_types = [convert_sp_to_sf_type(infer_type(arg)) for arg in args]
+            func_signature = f"{sproc_name.upper()}({', '.join(arg_types)})"
+            sproc_desc = self._run_query(f"describe procedure {func_signature}")
+            return_type = sproc_desc[1][1]
+            return return_type.upper().startswith("TABLE")
+        except Exception:
```

Comment:
what error are we try to catch here? when describe procedure fails? Should we raise an exception directly or have a warning?

Reply:
Yeah we should at least print a warning because here we are falling back to non-table sproc and will only return the first field, which could be surprising to user.

Reply:
added